#!/usr/bin/env python
PACKAGE = "graph_slam_tools"
import roslib;roslib.load_manifest(PACKAGE)
import math

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_height", double_t, 0, "The minimum height to include from the point cloud.",0.0, -100.0, 100.0)
gen.add("max_height", double_t, 0, "The maximum height to include from the point cloud.",1.0, -100.0, 100.0)
gen.add("angle_increment", double_t, 0, "The angle increment of the resulting laser scan.", math.pi/180.0/2.0, 0, math.pi)
gen.add("range_min", double_t, 0, "The minimum range of the resulting laser scan.", 0.45, 0.0, 100.0)
gen.add("range_max", double_t, 0, "The maximum range of the resulting laser scan.", 5.0, 0.0, 100.0)
gen.add("resolution", double_t, 0, "The grid resolution in m.", 0.1, 0.0, 100.0)
gen.add("recompute_laserscans", bool_t, 0, "true, if the laserscans should be recomputed from the depth images each time", False)

exit(gen.generate(PACKAGE, "graph_slam_tools", "OccupancyGridProjector"))
